staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -249,6 +263,12 @@ func New(lifecycleCtx context.Context, opts Options) (AuthenticatorTokenWithHeal
 		return nil, err
 	}
 
+	// require APIServerID to be non-empty because it is used in metrics
+	if len(opts.APIServerID) == 0 {
+		return nil, fmt.Errorf("oidc: APIServerID must be set")
+	}
```

Comment:
This will be a breaking change because we now required `APIServerID` to be set when using the oidc package as a library.

Reply:
Hmm actually the API server ID feature has been beta since v1.26 but it is not GA.  I think until it goes GA, we should probably skip these metrics altogether if the ID is empty.  We can require it once the API server ID feature is GA.

Reply:
Or maybe we set it to `none` until the API server ID feature is GA.

Reply:
setting it to `none` instead of skipping the metrics conditionally sgtm!

Reply:
pushed an update https://github.com/kubernetes/kubernetes/pull/123642/commits/6a0b77b94aff673004b5ab8d20ad6f742afc7a9b

1. Follow the same pattern as other places - if api server id is not set, it'll be empty in the metrics
2. Feature gate the metrics, so it doesn't impact users that use CLI flags to configure authenticator